### PR TITLE
Checklist improvements

### PR DESCRIPTION
**Pull request by neurogoo**
Add way to tag tasks. Also highlight if personio information is different than checklist info

### CODE REVIEW

File: checklist-app/src/Futurice/App/Checklist/Pages/Checklist.hs [hunk 1]
```diff
@@ -87,6 +89,9 @@ checklistPage world today authUser checklist = checklistPage_ (view nameText che
                     for_ (checklist ^? checklistTasks . ix prereqTid) $ \_ -> do
                         taskLink prereqTask
                         br_ []
+                td_ $ for_ (task ^. taskTags) $ \tag -> do
```
Comment (phadej) [inline checklist-app/src/Futurice/App/Checklist/Pages/Checklist.hs]: `ul_` + `li_` would be more correct semantically

File: checklist-app/src/Futurice/App/Checklist/Pages/CreateEmployee.hs [hunk 2]
```diff
@@ -59,12 +61,12 @@ personioToTemplate es e = Tmpl
     , tmplOffice       = e ^. Personio.employeeOffice
     , tmplTribe        = e ^. Personio.employeeTribe
     , tmplStartingDay  = e ^. Personio.employeeHireDate
+    , tmplEndDay       = e ^. Personio.employeeEndDate
```
Comment (phadej) [inline checklist-app/src/Futurice/App/Checklist/Pages/CreateEmployee.hs]: I wouldn't add `tmplEndDay` but rather parametrise how template is created from personio.

File: checklist-app/src/Futurice/App/Checklist/Pages/CreateEmployee.hs [hunk 3]
```diff
@@ -160,7 +163,7 @@ createEmployeePage world authUser memployee pemployee pes = checklistPage_ "Crea
             input_
                 [ futuId_ "employee-starting-day"
                 , type_ "date"
-                , value_ $ maybe "" toQueryParam (tmpl >>= tmplStartingDay)
+                , value_ $ maybe "" toQueryParam $ if leaving then tmpl >>= tmplEndDay else tmpl >>= tmplStartingDay
```
Comment (phadej) [inline checklist-app/src/Futurice/App/Checklist/Pages/CreateEmployee.hs]: i.e. use here only `startingDay`, (it's unfortunate name, as it's not really startingDay but "zero hour"

File: checklist-app/src/Futurice/App/Checklist/Pages/Employee.hs [hunk 4]
```diff
@@ -160,5 +174,51 @@ employeePage world authUser employee _personios = checklistPage_ (view nameText
...
+            Just P.PermanentAllIn -> Just ContractTypePermanent
+            Just P.FixedTerm      -> Just ContractTypeFixedTerm
+            Just P.Permanent      -> Nothing -- TODO!
+
+    hasDifferentPersonioInfo :: Maybe Text -> Maybe Text -> Text -> HtmlT Identity ()
+    hasDifferentPersonioInfo info Nothing t = wrapToWarningLabel $ personioText info t
+    hasDifferentPersonioInfo info locali t = if info == locali then "" else wrapToWarningLabel $ personioText info t
+
+    hasDifferentPersonioPhone :: Maybe P.Employee -> Maybe Text -> HtmlT Identity ()
+    hasDifferentPersonioPhone info locali = maybe (pure ()) (\x -> hasDifferentPersonioInfo (x ^. P.employeeHomePhone) locali "phone") info
```
Comment (phadej) [inline checklist-app/src/Futurice/App/Checklist/Pages/Employee.hs]: `maybe (pure ()) (\x -> ...) something` = `for_ something $ \x -> ...`

File: checklist-app/src/Futurice/App/Checklist/Pages/Personio.hs [hunk 5]
```diff
@@ -84,7 +84,7 @@ employeeTable hire world employees = fullRow_ $ table_ $ do
         td_ $ button_
             [ class_ "button"
             , data_ "futu-link-button" $ linkToText
-            $ safeLink checklistApi createEmployeePageEndpoint Nothing (e ^? Personio.employeeId)
+            $ safeLink checklistApi createEmployeePageEndpoint Nothing (e ^? Personio.employeeId) $ if hire then False else True
```
Comment (phadej) [inline checklist-app/src/Futurice/App/Checklist/Pages/Personio.hs]: `hire` ?

File: checklist-app/src/Futurice/App/Checklist/Pages/Tasks.hs [hunk 6]
```diff
@@ -78,6 +80,9 @@ tasksPage world authUser@(_fu, _viewerRole) mrole mlist =
                 td_ $ forOf_ (taskPrereqs . folded . getter (\tid' -> world ^. worldTasks . at tid') . _Just) task $ \prereqTask -> do
                     taskLink prereqTask
                     br_ []
+                td_ $ for_ (task ^. taskTags) $ \tag -> do
```
Comment (phadej) [inline checklist-app/src/Futurice/App/Checklist/Pages/Tasks.hs]: `ul_ + li_`

File: checklist-app/src/Futurice/App/Checklist/Pages/Personio.hs (hunk 5 above)
Comment (neurogoo) [reply checklist-app/src/Futurice/App/Checklist/Pages/Personio.hs]: I guess I could have used hire in the CreateEmployee, but I leaving variable felt more natural

File: checklist-app/src/Futurice/App/Checklist/Pages/Employee.hs [hunk 7]
```diff
@@ -160,5 +174,50 @@ employeePage world authUser employee _personios = checklistPage_ (view nameText
...
+            Just P.PermanentAllIn -> Just ContractTypePermanent
+            Just P.FixedTerm      -> Just ContractTypeFixedTerm
+            Just P.Permanent      -> Nothing -- TODO!
+
+    hasDifferentPersonioInfo :: Maybe Text -> Maybe Text -> Text -> HtmlT Identity ()
+    hasDifferentPersonioInfo info Nothing t = wrapToWarningLabel $ personioText info t
+    hasDifferentPersonioInfo info locali t = if info == locali then "" else wrapToWarningLabel $ personioText info t
+
+    hasDifferentPersonioPhone :: Maybe P.Employee -> Maybe Text -> HtmlT Identity ()
+    hasDifferentPersonioPhone info locali = for_ info $ \x -> hasDifferentPersonioInfo (x ^. P.employeeHomePhone) locali "phone"
```
Comment (phadej) [inline checklist-app/src/Futurice/App/Checklist/Pages/Employee.hs]: A lot of copy&paste code. OK for now, but have to think how to make this prettier in the future

**Pull request merged**